file-proxy examples

kbrandwijk opened the pull request:
Set of examples around a webtask-based wrapper for the File API, including a bigger authentication example. To follow on the 'practice what you preach', I changed the first example to use graphql-request. The other examples use [request](https://github.com/request/request).

kbrandwijk commented:
I have now included the full authenticated file proxy example. The file-proxy example set is now complete.

kbrandwijk commented:
To follow on the 'practice what you preach', I changed the first example to use graphql-request.

marktani commented:
Currently reviewing this example, it's great! 🙏 
When I upload a `png` file to the `simple-file-proxy` and visit its `newUrl`, I get the following log in the DevTools:

> Resource interpreted as Document but transferred with MIME type image/png: "https://wt-nilan_marktanner-gmail_com-0.run.webtask.io/simple/cj49x05nvjdeh01735bvg73hx/cj49xo50v03um0168cr63t9r0".

Do you have any insights into this?

marktani commented on file-proxy/README.md:
````diff
@@ -0,0 +1,175 @@
+```
+Now, try to upload more than one file at once to your endpoint:
+```sh
+curl -X POST '<webtask endpoint url>/__PROJECT_ID__' -F "file1=@example.png;filename=myname.png;file2=@example2.png;filename=testfile.png"
````
I could not upload multiple files. Instead, I received the following warning:

> Warning: skip unknown form field: file2=@example2.png

And only the *first* file will be uploaded, while being named `testfile.png`.

kbrandwijk replied:
The syntax of the curl command is wrong. It requires to seperate -F sections. I will update the readme.

kbrandwijk commented:
It might be a header thing. I'll check it when I'm back behind the computer.

marktani commented on file-proxy/auth-file-proxy/auth-file-proxy.js:
```diff
@@ -0,0 +1,136 @@
+      var resource = request.get(response.data.MyFile.file.url);
+
+      // As soon as we get a response, we copy the headers
+      // Content-length is removed, because the decrypted file does not have the same length
```
is it feasible to adjust the `content-length` header with the new length, instead?

marktani commented:
I have verified the following examples to work:

* `simple-file-proxy`
* `extension-file-proxy`

I found problems with `multiple-file-proxy`, as detailed above.
I read the other examples thoroughly and they look great as well 👍

kbrandwijk commented:
@marktani, I updated the readme with the correct curl syntax for the multi-file-upload, I fixed the auth-file-proxy content-length issue (by storing/retrieving original file size in `MyFile`).
The warning from Chrome is because I proxy the original request, it's only a warning, and it might be able to remove with additional tweaking, but that's outside of the scope of this example.

yusinto commented on file-proxy/README.md:
````diff
@@ -0,0 +1,211 @@
+Your webtask is now deployed! You can upload a file using any tool that support `multipart/form-data`, like Postman, or the command-line tool `curl`:
+
+```sh
+curl -X POST '<webtask endpoint url>/__PROJECT_ID__' -F "data=@small-logo.png;filename=small.png"
````
Nit: use single quotes throughout to be consistent:
curl -X POST '<webtask endpoint url>/__GRAPHCOOL_PROJECT_ID__' -F 'data=@small-logo.png;filename=small.png'

yusinto commented on file-proxy/simple-file-proxy.js:
```diff
@@ -0,0 +1,75 @@
+"use latest"
```
I have made some minor changes to this file, perhaps it's easier to view these changes directly [here](https://github.com/yusinto/functions/blob/master/file-proxy/simple-file-proxy.js)

kbrandwijk commented:
@yusinto I processed your code styling suggestions

yusinto commented on file-proxy/encrypted-file-proxy.js:
```diff
@@ -0,0 +1,98 @@
+const app = express();
+
+// The upload endpoint
+app.post('/:projectid', (req, res) => {
```
I think you mean `projectId`? :)
I also have made some minor adjustments and corrections [here](https://github.com/yusinto/functions/blob/master/file-proxy/encrypted-file-proxy.js)

kbrandwijk replied:
Something is definitely off with that change tracking when you make changes. It's easier to commit them to the PR like @marktani did above. Anyways, the projectId was one of the things I overlooked having to do a manual merge of your previous changes... I will update that. What other things did you change? Only the use of graphql-request in encrypted-file-proxy?

yusinto commented:
Those two are the biggest ones. I also replaced double quotes with single
to be consistent and changed string concatenation to string templating.

Apologies about the way I use github! I'm reviewing auth-file-proxy and
i'll try and submit a PR for that to make it easier.

On Mon, Jun 26, 2017 at 9:30 PM, kbrandwijk <notifications@github.com>
wrote:

> *@kbrandwijk* commented on this pull request.
> ------------------------------
>
> In file-proxy/encrypted-file-proxy.js
> <https://github.com/graphcool-examples/functions/pull/11#discussion_r123986693>
> :
>
> > @@ -0,0 +1,98 @@
> +"use latest"
> +
> +import Webtask from 'webtask-tools';
> +import express from 'express';
> +import crypto from 'crypto';
> +import multiparty from 'multiparty';
> +import formData from 'form-data';
> +import request from 'request';
> +
> +const app = express();
> +
> +// The upload endpoint
> +app.post('/:projectid', (req, res) => {
>
> Something is definitely off with that change tracking when you make
> changes. It's easier to commit them to the PR like @marktani
> <https://github.com/marktani> did above. Anyways, the projectId was one
> of the things I overlooked having to do a manual merge of your previous
> changes... I will update that. What other things did you change? Only the
> use of graphql-request in encrypted-file-proxy?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/graphcool-examples/functions/pull/11#discussion_r123986693>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABhO9aUmAjbGPH14jNvogkLaNsCRqxfSks5sH5ZIgaJpZM4N9YV1>
> .
>

kbrandwijk commented:
My last commit is not showing up here...

marktani commented:
> PR and history completely screwed, unable to squash. Please squash on PR merge...

😞

marktani commented:
Thanks so much @kbrandwijk for that amazing contribution! 🙏 Also thanks to @yusinto for the help reviewing this. Next time let's make sure the commits are not messed up 😆
